### PR TITLE
Increase Shapely and GDAL versions

### PR DESCRIPTION
**Pull request by dcdenu4**
This PR mostly deals with bumping the Shapely version and GDAL version.

**Shapely and GDAL version bump**
- `shapely>=1.7.0,<1.8.0`
- `gdal==3.1.2`
- Add bundling of `libgeos_*.dylib` for MacOS in `invest.spec`

`shapely 1.7.0` requires `geos >= 3.8.1`
`gdal 3.1.2` requires `libgdal 3.1.2` which requires `proj>=7.0.0`. 

**Fixed tests for `proj>=7.0.0`**
These version bumps include a dependency on PROJ 7+. PROJ 7 makes some changes that affect some of our tests https://proj.org/news.html
- Updates to wind energy tests
- Updates to urban cooling tests
- Update to wind energy regression tests here: https://bitbucket.org/natcap/invest-test-data/branch/task/gdal312-proj7
- PR for changed test data is here: https://bitbucket.org/natcap/invest-test-data/pull-requests/17/update-wind-regression-tests-for-proj-7

**Other Bonus Changes**
- `setup-miniconda` action has changed repo url
- Fix a few tests that were not cleaning up created test files.

### CODE REVIEW

File: exe/invest.spec [hunk 1]
```diff
@@ -48,6 +48,10 @@ if is_darwin:
     a.binaries += [
         (os.path.basename(name), name, 'BINARY') for name in
         glob.glob(os.path.join(conda_env, 'lib', 'libspatialindex*.dylib'))]
+    # add shapely dependency geos from conda environment
+    a.binaries += [
+        (os.path.basename(name), name, 'BINARY') for name in
+        glob.glob(os.path.join(conda_env, 'lib', 'libgeos_*.dylib'))]
```
Comment (phargogh) [inline exe/invest.spec:54]: Whoops, thanks for adding this @dcdenu4 !

File: requirements.txt [hunk 2]
```diff
@@ -12,13 +12,13 @@
 # scripts/convert-requirements-to-conda-yml.py as though it can only be found
 # on pip.
 
-GDAL>=3.0.4,<=3.1.3
+GDAL==3.1.2
```
Comment (phargogh) [inline requirements.txt:15]: Does GDAL>3.1.2 break the build?

**Review by phargogh — COMMENTED** [review]
Thanks @dcdenu4 !  I just had a question about the GDAL version pinning, but this looks good otherwise!

File: requirements.txt (hunk 2 above)
Comment (phargogh) [reply requirements.txt:15]: And more generally, can this upper bound be left open (or at least have a version cap set farther out)?

Comment (dcdenu4) [reply requirements.txt:15]: Fair question, let me check GDAL's requirements for other versions!

Comment (dcdenu4) [reply requirements.txt]: Hey @phargogh, it looks like `libgdal 3.1.2` has builds for `PROJ 7.0.0 to 7.1.1` and `libgdal 3.1.3` requires `PROJ 7.1.1+`. I've actually been testing with 7.1.1. Anyway, I think opening the upper bound is fine for now. https://github.com/natcap/invest/pull/350/commits/d823912c0bc045ea7663e67854ef69439548300b

**Review by phargogh — APPROVED** [review]
Sounds good to me!

For future us, the mac builds are still failing, but this is still a homebrew/github issue that is outside of our control.

**Pull request merged**